repair/row_level.cc

```diff
@@ -1855,7 +1855,7 @@ class repair_meta {
             repair_hash_set set_diff,
             needs_all_rows_t needs_all_rows,
             locator::host_id remote_node, unsigned node_idx,
-            const locator::effective_replication_map& erm, bool small_table_optimization,
+            locator::effective_replication_map_ptr erm, bool small_table_optimization,
```

Comment:
I think such coupling between parameters is best to avoid.
How about:
```c++
struct small_table_optimization_params {
    const locator::effective_replication_map& erm;
};
```
Then replace both params with just `std::optional<small_table_optimization_params>`. This way we keep our cake and eat it too: we avoid passing erm when not using small table optimization and it is impossible to make a mistake.